### PR TITLE
test: Add unit tests for newly added components in sms to submit receipt

### PR DESCRIPTION
**Pull request by yksolanki9**
Items covered in this PR: 
- Unit tests for mobile number verification service:  https://app.clickup.com/t/85zt2xb53
- Unit tests for info card component:  https://app.clickup.com/t/85zt2xb1b
- Unit tests for popup with bullets component: https://app.clickup.com/t/85zt2xb2e

Code Coverage:
Screenshot 2023-05-18 at 3 48 01 PM

Screenshot 2023-05-18 at 4 44 32 PM

Screenshot 2023-05-18 at 5 15 13 PM

### CODE REVIEW

File: src/app/core/services/mobile-number-verification.service.spec.ts [hunk 1]
```diff
@@ -1,17 +1,52 @@
...
+    });
+  });
+
+  it('verifyOtp(): should verify otp', () => {
+    const otp = '123456';
+    service.verifyOtp(otp).subscribe(() => {
+      expect(apiService.post).toHaveBeenCalledOnceWith(
+        '/orgusers/check_mobile_verification_code',
+        otp,
+        jasmine.any(Object)
```
Comment (yksolanki9) [inline src/app/core/services/mobile-number-verification.service.spec.ts:43]: Comparing this directly with HttpHeader does not work, so checking the specific header separately

File: src/app/fyle/my-profile/info-card/info-card.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,56 @@
...
     expect(component).toBeTruthy();
   });
+
+  it('copyToClipboard(): Should copy content to clipboard and emit copiedText event', fakeAsync(() => {
+    spyOn(component, 'copyToClipboard').and.callThrough();
+    clipboardService.writeString.and.resolveTo();
+    spyOn(component.copiedText, 'emit');
+
+    const infoCard = fixture.debugElement.query(By.css('.info-card__card'));
+    infoCard.nativeElement.click();
```
Comment (rvab) [inline src/app/fyle/my-profile/info-card/info-card.component.spec.ts:41]: there must be utility functions for these. Please check.
* getting an element and its text content
* clicking an element
etc

File: src/app/fyle/my-profile/info-card/info-card.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,56 @@
...
     fixture.detectChanges();
   }));
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('copyToClipboard(): Should copy content to clipboard and emit copiedText event', fakeAsync(() => {
+    spyOn(component, 'copyToClipboard').and.callThrough();
+    clipboardService.writeString.and.resolveTo();
```
Comment (rvab) [inline src/app/fyle/my-profile/info-card/info-card.component.spec.ts:37]: what does `resolveTo` do?

File: src/app/core/services/mobile-number-verification.service.spec.ts (hunk 1 above)
Comment (rvab) [inline src/app/core/services/mobile-number-verification.service.spec.ts:43]: Can you not do the following
```
    const headers = new HttpHeaders({ 'Content-Type': 'application/json; charset=utf-8' });

```

File: src/app/core/services/mobile-number-verification.service.spec.ts [hunk 4]
```diff
@@ -1,17 +1,52 @@
...
+      );
+
+      //Check if the 3rd argument is an object with Content-Type header set to desired value
+      const calls = apiService.post.calls.all();
+      const data = calls[0].args[2] as { headers: HttpHeaders };
+      expect(data.headers.get('Content-Type')).toBe('application/json; charset=utf-8');
```
Comment (rvab) [inline src/app/core/services/mobile-number-verification.service.spec.ts:49]: I didn't understand why these lines from 47-49 are required.

File: src/app/fyle/my-profile/info-card/info-card.component.spec.ts (hunk 3 above)
Comment (yksolanki9) [reply src/app/fyle/my-profile/info-card/info-card.component.spec.ts:37]: https://jasmine.github.io/api/4.6/SpyStrategy.html#:~:text=3.5.0-,resolveTo(value),-Tell%20the%20spy

Returns a promise and resolves it with the passed argument

File: src/app/core/services/mobile-number-verification.service.spec.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/core/services/mobile-number-verification.service.spec.ts:43]: This does not work, functions are compared by reference. So, the assertion fails with message `Expected function to equal function`. And so, in the next assertion, I just check the value of the header

File: src/app/core/services/mobile-number-verification.service.spec.ts (hunk 4 above)
Comment (yksolanki9) [reply src/app/core/services/mobile-number-verification.service.spec.ts:49]: Functions are compared by reference. So, we cannot check for equality by created a new header. So, here I'm checking if content-type header is same as the one we expected

File: src/app/core/services/mobile-number-verification.service.spec.ts [hunk 5]
```diff
@@ -1,17 +1,52 @@
...
-
+import { of } from 'rxjs';
+import { HttpHeaders } from '@angular/common/http';
 import { MobileNumberVerificationService } from './mobile-number-verification.service';
+import { ApiService } from './api.service';
 
-//TODO: Write unit tests for this service
-xdescribe('MobileNumberVerificationService', () => {
+describe('MobileNumberVerificationService', () => {
   let service: MobileNumberVerificationService;
```
[the remark this answers is not in the document]
Comment (yksolanki9) [reply src/app/core/services/mobile-number-verification.service.spec.ts:8]: Done

File: src/app/fyle/my-profile/info-card/info-card.component.spec.ts [hunk 6]
```diff
@@ -1,25 +1,57 @@
...
+    expect(clipboardService.writeString).toHaveBeenCalledOnceWith(component.contentToCopy);
+    expect(component.copiedText.emit).toHaveBeenCalledOnceWith(component.toastMessageContent);
+  }));
+
+  it('should render title and content in the ui', () => {
+    const title = getElementBySelector(fixture, '.info-card__card__title');
+    const content = getElementBySelector(fixture, '.info-card__content-container__content');
+
+    expect(title.textContent).toEqual(component.title);
+    expect(content.textContent).toEqual(component.content);
```
Comment (rvab) [inline src/app/fyle/my-profile/info-card/info-card.component.spec.ts:55]: there's a utility function for this. please use it.

File: src/app/shared/components/popup-with-bullets/popup-with-bullets.component.spec.ts [hunk 7]
```diff
@@ -1,24 +1,121 @@
...
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('dimissPopover(): should dismiss popover', () => {
+    spyOn(component, 'dismissPopover').and.callThrough();
+    popoverController.dismiss.and.resolveTo();
+
+    const dismissCta = fixture.debugElement.query(By.css('.popup-with-bullets__cta'));
```
Comment (rvab) [inline src/app/shared/components/popup-with-bullets/popup-with-bullets.component.spec.ts:76]: use utility function

File: src/app/shared/components/popup-with-bullets/popup-with-bullets.component.spec.ts [hunk 8]
```diff
@@ -1,24 +1,121 @@
...
+    expect(component.dismissPopover).toHaveBeenCalledOnceWith();
+    expect(popoverController.dismiss).toHaveBeenCalledOnceWith();
+  });
+
+  it('copyToClipboard(): Should copy content to clipboard and show toast message', () => {
+    clipboardService.writeString.and.resolveTo();
+    spyOn(component, 'copyToClipboard').and.callThrough();
+    spyOn(component, 'showToastMessage');
+
+    const infoCard = fixture.debugElement.query(By.css('.popup-with-bullets__list-item__copy-text'));
```
Comment (rvab) [inline src/app/shared/components/popup-with-bullets/popup-with-bullets.component.spec.ts:88]: same as above

**Review by rvab — APPROVED** [review]
Use utility function

**Pull request merged**